api/argo-templates/ncn/worker.post-rebuild.yaml

```diff
@@ -33,5 +33,69 @@ tasks:
           value: "{{ `{{inputs.parameters.dryRun}}` }}"
         - name: scriptContent
           value: |
-            TODO
+            export TOKEN=$(curl -k -s -S -d grant_type=client_credentials \
+            -d client_id=admin-client \
+            -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` \
+            https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+            csi handoff bss-update-param --set metal.no-wipe=1 --limit $(ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null {{ `{{inputs.parameters.targetNcn}}` }}
+  - name: wait-for-cfs-after-rebuild
+    templateRef:
+      name: ssh-with-kubeconfig
+      template: bash-script-example
+    arguments:
+      parameters:
+        - name: dryRun
+          value: "{{ `{{inputs.parameters.dryRun}}` }}"
+        - name: scriptContent
+          value: |
+            /usr/share/doc/csm/upgrade/1.2/scripts/cfs/wait_for_configuration.sh \
+              --xnames \
+              $(ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null {{ `{{inputs.parameters.targetNcn}}` }} 'cat /etc/cray/xname')
+  - name: wait-for-key-pods
+    templateRef:
+      name: ssh-with-kubeconfig
+      template: bash-script-example
+    arguments:
+      parameters:
+        - name: dryRun
+          value: "{{ `{{inputs.parameters.dryRun}}` }}"
+        - name: scriptContent
+          value: |
+            while true; do
+              output=$(kubectl get po -A -o wide | grep -e etcd -e speaker | grep {{ `{{inputs.parameters.targetNcn}}` }} | awk '{print $4}')
+              if [ ! -n "$output" ]; then
+                #
+                # No pods scheduled to start on this node, we are done
+                #
+                break
+              fi
+              set +e
```

Comment:
Could just put this block in the [[ ]] in the if statement, anything that can fail in a test block isn't covered by set +/-e will reduce the amount of shell in use.

aka just change this to 
```
if echo "$output" | grep -Ev '(Running|Completed)' > dev/null; then
 ...
```

And can remove the unneeded set/etc.... entirely.

Example of what I mean in general:

```
set -eu
if echo foo | grep bar > /dev/null 2>&1; then
    echo nope
else
    echo ok
fi

set +e
echo foo | grep bar > /dev/null 2>&1
echo $?
set -e

echo foo | grep bar > /dev/null 2>&1
echo $?
```

Produces:
```
/tmp/test.sh
ok
1
```

As that last echo is ignored, but as you can see the if block doesn't care even when set -e is present, reduces line count as well which I think helps long term.